### ISSUE TITLE
The example ActiveMQArtemisSecurity definition doesn't create any users

### ISSUE THREAD

**Issue by monodot** [artemiscloud/activemq-artemis-operator#105]
**Describe the bug**
When I try to apply security configuration using the example ActiveMQArtemisSecurity CR YAML `deploy/crs/broker_activemqartemissecurity_cr.yaml` which is included in the distribution, nothing happens on my deployed Artemis broker. I don't see any user created in the properties file, and I can't authenticate as user 'bob'.

Am I missing something in my setup / CR?

**To Reproduce**
1. Download the 0.20 release (`curl -OL https://github.com/artemiscloud/activemq-artemis-operator/archive/refs/tags/v0.20.0.tar.gz`)
2. Follow the steps in the docs to install the CRDs and Operator, also installing the ActiveMQArtemisSecurity CRD.
3. Create this broker:
```
apiVersion: broker.amq.io/v2alpha5
kind: ActiveMQArtemis
metadata:
  name: mybroker
spec:
  deploymentPlan:
    size: 1
    image: placeholder
    requireLogin: true
```
4. Create this ActiveMQArtemisSecurity CR:
```
apiVersion: broker.amq.io/v1alpha1
kind: ActiveMQArtemisSecurity
metadata:
  name: broker-security-config
spec:
  applyToCrNames:
    - "*"
  loginModules:
    propertiesLoginModules:
    - name: 'prop-module'
      users:
      - name: bob
        password: mysecret
        roles:
        - root
  securityDomains:
    brokerDomain:
      name: 'activemq'
      loginModules:
      - name: 'prop-module'
        flag: 'sufficient'
```

4. Try to put a message on a queue. The auto-generated admin user can authenticate, but the named user can't:

```
/ # java -jar /a/a.jar --artemis-core --user gkyW8Hrs --pass 7KBeh41j --broker tcp://mybroker-hdls-svc:61616 --put "foobar" sandwiches
Message sent
Operation completed in 153ms (excluding connect)

/ # java -jar /a/a.jar --artemis-core --user bob --pass mysecret --broker tcp://mybroker-hdls-svc:61616 --put "foobar" sandwiches
javax.jms.JMSSecurityException: AMQ229031: Unable to validate user from /10.128.2.83:57868. Username: bob; SSL certificate subject DN: unavailable
```

Only the admin user exists in the users properties file:

```
$ cat /home/jboss/amq-broker/etc/artemis-users.properties
...
gkyW8Hrs = ENC(1024:8D873D1EDFB8ABACBC1A8229AC6A691B2856427B385167AC5DA636A8B0D0CF7C:50DE2FF97B69C8B8D127356F8F1090730F8F805DAA288E804E199788394EF0C9363671038857CB7F83ACE9022ACE2119792B9FCFB72CA68D026A5458B2D4C7CF)
```

**Platform (please complete the following information):**
 - Operator version: 0.20
 - OpenShift v4.6.20

The Broker pod is running this image:

```
$ oc describe pod mybroker-ss-0 | grep Image
    Image:         quay.io/artemiscloud/activemq-artemis-broker-init:0.2.7
    Image ID:      quay.io/artemiscloud/activemq-artemis-broker-init@sha256:878acaa33b89ff72892e4305e688bf1da7196e84a79981f27b5ced62e3208fc5
    Image:         quay.io/artemiscloud/activemq-artemis-broker-kubernetes:0.2.5
    Image ID:      quay.io/artemiscloud/activemq-artemis-broker-kubernetes@sha256:101abbc3c2c2da662fddd508573a447aaac90155971b383374bc3bbfc6b755ff
```

The Operator is running this image:

```
$ oc describe pod activemq-artemis-operator-77777444fc-t75gm | grep Image
    Image:         quay.io/artemiscloud/activemq-artemis-operator:0.20.0
    Image ID:      quay.io/artemiscloud/activemq-artemis-operator@sha256:86181b1b5a2dc9dc9089f4ba6b28c1e971cb7d02dbf7c09e4e93b6b9d5462135

```

**Expected behavior**
The Security controller should create users, and apply some sort of security config on the broker.
I should be able to log in as the user in the SecuritySettings CR. 

I see this in the controller logs, when the Security controller is reconciling:

```
2021-09-27T16:02:47.112Z	INFO	controller_v1alpha1activemqartemissecurity	Reconciling ActiveMQArtemisSecurity	{"Request.Namespace": "sso-app-demo", "Request.Name": "broker-security"}
2021-09-27T16:02:47.112Z	INFO	controller_v1alpha1activemqartemissecurity	Fetched instance	{"Request.Namespace": "sso-app-demo", "Request.Name": "broker-security", "the instance": {"kind":"ActiveMQArtemisSecurity","apiVersion":"broker.amq.io/v1alpha1","metadata":{"name":"broker-security","namespace":"sso-app-demo","selfLink":"/apis/broker.amq.io/v1alpha1/namespaces/sso-app-demo/activemqartemissecurities/broker-security","uid":"362f5aae-d42f-4caf-8d20-656e6744641d","resourceVersion":"293655","generation":1,"creationTimestamp":"2021-09-27T16:02:47Z","annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"broker.amq.io/v1alpha1\",\"kind\":\"ActiveMQArtemisSecurity\",\"metadata\":{\"annotations\":{},\"name\":\"broker-security\",\"namespace\":\"sso-app-demo\"},\"spec\":{\"loginModules\":{\"propertiesLoginModules\":[{\"name\":\"prop-module\",\"users\":[{\"name\":\"bob\",\"password\":\"mysecret\",\"roles\":[\"root\"]}]}]},\"securityDomains\":{\"brokerDomain\":{\"loginModules\":[{\"flag...
2021-09-27T16:02:47.112Z	INFO	controller_v2alpha5activemqartemis	A config handler has been added	{"new handler": {"SecurityCR":{"kind":"ActiveMQArtemisSecurity","apiVersion":"broker.amq.io/v1alpha1","metadata":{"name":"broker-security","namespace":"sso-app-demo","selfLink":"/apis/broker.amq.io/v1alpha1/namespaces/sso-app-demo/activemqartemissecurities/broker-security","uid":"362f5aae-d42f-4caf-8d20-656e6744641d","resourceVersion":"293655","generation":1,"creationTimestamp":"2021-09-27T16:02:47Z","annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"broker.amq.io/v1alpha1\",\"kind\":\"ActiveMQArtemisSecurity\",\"metadata\":{\"annotations\":{},\"name\":\"broker-security\",\"namespace\":\"sso-app-demo\"},\"spec\":{\"loginModules\":{\"propertiesLoginModules\":[{\"name\":\"prop-module\",\"users\":[{\"name\":\"bob\",\"password\":\"mysecret\",\"roles\":[\"root\"]}]}]},\"securityDomains\":{\"brokerDomain\":{\"loginModules\":[{\"flag\":\"sufficient\",\"name\":\"prop-module\"}],\"name\...
2021-09-27T16:02:47.113Z	DEBUG	kubebuilder.controller	Successfully Reconciled	{"controller": "v1alpha1activemqartemissecurity-controller", "request": "sso-app-demo/broker-security"}
```

Judging by the logs above, it seems to miss executing this func entirely (I would expect to see some line like "Reconciling security" in the logs):

https://github.com/artemiscloud/activemq-artemis-operator/blob/v0.20.0/pkg/controller/broker/v1alpha1/activemqartemissecurity/activemqartemissecurity_controller.go#L201

Is this a bug? Or is there something else that I need to add to the Broker CR, or the Operator, to get this particular function to run (the bit that does the reconciling)?

Thanks!

**Comment by gaohoward** [comment]
@monodot Hi, on the first look your CR seems ok, but it probably need more configure on permissions for user bob.
 can you post your broker pod's log? also can you post your operator.yaml?

thanks!

**Comment by monodot** [comment]
Thanks for the pointers @gaohoward . I've given bob the role `admin` now, so the ActiveMQArtemisSecurity looks like this:

```
apiVersion: broker.amq.io/v1alpha1
kind: ActiveMQArtemisSecurity
metadata:
  name: broker-security-config
spec:
  loginModules:
    propertiesLoginModules:
    - name: 'prop-module'
      users:
      - name: bob
        password: mysecret
        roles:
        - root
        - admin
  securityDomains:
    brokerDomain:
      name: 'activemq'
      loginModules:
      - name: 'prop-module'
        flag: 'sufficient'
```

The _operator.yaml_ that I'm using is [the one in the v0.20 tag](https://github.com/artemiscloud/activemq-artemis-operator/blob/v0.20.0/deploy/operator.yaml):

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: activemq-artemis-operator
spec:
  replicas: 1
  selector:
    matchLabels:
      name: activemq-artemis-operator
  strategy:
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 1
    type: RollingUpdate
  template:
    metadata:
      labels:
        name: activemq-artemis-operator
    spec:
      containers:
      - args:
        # Explicitly set the logging level.
        # Valid values are debug, info, and error
        # from most to least.
        # If running entrypoint_debug then use '-- --zap-level debug'
        - '--zap-level debug'
        - '--zap-encoder console'
        command:
        - /home/activemq-artemis-operator/bin/entrypoint
        env:
        - name: OPERATOR_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.labels['name']
        - name: POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
          # to watch all namespaces use '*' or empty ''
          # to watch multiple namespaces use comma separated list 'ns1,ns2'
          # default is to watch the operator's namespace (backward compatibility)
        - name: WATCH_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace

        - name: RELATED_IMAGE_ActiveMQ_Artemis_Broker_Init_2150
          value: quay.io/artemiscloud/activemq-artemis-broker-init:0.2.2
        - name: RELATED_IMAGE_ActiveMQ_Artemis_Broker_Init_2160
          value: quay.io/artemiscloud/activemq-artemis-broker-init:0.2.4
        - name: RELATED_IMAGE_ActiveMQ_Artemis_Broker_Init_2180
          value: quay.io/artemiscloud/activemq-artemis-broker-init:0.2.7
        - name: RELATED_IMAGE_ActiveMQ_Artemis_Broker_Kubernetes_2150
          value: quay.io/artemiscloud/activemq-artemis-broker-kubernetes:0.2.0
        - name: RELATED_IMAGE_ActiveMQ_Artemis_Broker_Kubernetes_2160
          value: quay.io/artemiscloud/activemq-artemis-broker-kubernetes:0.2.1
        - name: RELATED_IMAGE_ActiveMQ_Artemis_Broker_Kubernetes_2180
          value: quay.io/artemiscloud/activemq-artemis-broker-kubernetes:0.2.5

        image: quay.io/artemiscloud/activemq-artemis-operator:0.20.0
        imagePullPolicy: Always
        name: activemq-artemis-operator
        resources: {}
      serviceAccountName: activemq-artemis-operator
```

But, still the 'bob' user doesn't seem to be created:

```
sh-4.4$ cat /home/jboss/amq-broker/etc/artemis-users.properties 
vGYO2WgB = ENC(1024:21E284BCD3B437BB333CF......9DA850689ABA924732D7F318CF0B31)sh-4.4$ 
```

**Broker Pod log**
Here's the logs from the broker pod.
[broker-pod-log.txt](https://github.com/artemiscloud/activemq-artemis-operator/files/7242657/broker-pod-log.txt)

Here's the logs from the init-container.
[init-container.log](https://github.com/artemiscloud/activemq-artemis-operator/files/7242828/init-container.log)

**Comment by gaohoward** [comment]
@monodot Hey looks like the security doesn't get applied. We have a few issues fixed after v0.20.0, one of them is that if you apply the security CR after the broker CR it wont apply. Did you happen to apply the security CR **after** the broker CR has been deployed? If so try to deploy first the security CR then the broker CR.

**Comment by brusdev** [comment]
This issue will be closed because it has not been updated for a long time.
